books/spiders/spider.py

```diff
@@ -0,0 +1,62 @@
+import requests
+import scrapy
+from bs4 import BeautifulSoup
+from scrapy.http import Response
+
+
+class SpiderSpider(scrapy.Spider):
+    name = "spider"
+    allowed_domains = ["books.toscrape.com"]
+    start_urls = ["https://books.toscrape.com"]
+
+    def _parse_single_book(self, url: str) -> dict:
+        page = requests.get(url).content
```

Comment:
The `_parse_single_book` method uses `requests.get` to fetch the page content. In Scrapy, you should use Scrapy's request methods to fetch pages, as this allows Scrapy to manage requests and responses efficiently.